doc/adr/0001-record-architecture-decisions.md

```diff
@@ -0,0 +1,149 @@
+# ADR 1: Record architecture decisions
+
+Date: 2022-06-08
+
+## Status
+
+Draft
+
+## Context
+
+We are in search for a means to document our architectural and design decisions
+for all of our components.
+In order to do that, there is practice called architectural design records, ADR,
+that we can integrate into our workflow.
+
+This does not replace actual architecture documentation, but provides contributors:
+
+* the means to understand architectural and design decisions that were made
+* a framework for proposing changes to the current architecture
+
+For each decision, it is important to consider and remember:
+
+* what we've decided to do,
+* why we made this decision,
+* what is the impact of this decision,
+* what we learnt in the process.
+
+As we're already using rST, Sphinxdoc and readthedocs, it would be practical to
+integrate these ADRs as part of our current documentation infrastructure.
+
+## Decision
+
+* We will use _Architecture Decision Records_ to document, propose and discuss
+any important or significant architectural and design decisions.
+
+* The ADR format will follow one of the two formats described in the sections below.
+
+* We will follow the convention of storing those ADRs as Markdown formatted
+documents stored under `docs/adr` directory, as exemplified in Nat Pryce's
+[adr-tools](https://github.com/npryce/adr-tools). This does not imply we will
+be using `adr-tools` itself, as we might in the future diverse to the proposed structure.
+
+* We will keep rejected ADRs
+
+## Consequences
+
+_Architecture Decision Records_ should be written using the template described below which comes from
+Chapter 6.5.2 (_A Template for Documenting Architectural Decisions_) of
+_Documenting Software Architectures: Views and Beyond (2nd Edition)_.
+
+However, the mandatory sections are *Title*, *Issue*, *Decision*, *Status*, *Implications*.
+The rest are optional.
+
+Another good reference is _Architecture Decision Records_ as described by Michael Nygard in
+this [article](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions).
+
+### ADR template
+
+_Architecture Decision Records_ follow the template provided in
+Chapter 6.5.2 (_A Template for Documenting Architectural Decisions_) of
+_Documenting Software Architectures: Views and Beyond (2nd Edition)_.
+
+Here's the format (adapted from the book).
+
```

Comment:
For the list of items below, it could be cool to put these into a table, with the terms displaying in column one and the description in column two. 

| Term    | Description | 
| ------- | ----------- | 
| Title   | These documents have names that are short noun phrases. For example, “ADR 1: Deployment on Ruby on Rails 3.0.10” or “ADR 9: LDAP for Multitenant Integration” | 
| Issue (or context) | This section describes the architectural design issue being addressed. This description should leave no questions as to why this issue needs to be addressed now. The language in this section is value-neutral. It is simply describing facts. | 

etc.